### PR TITLE
LOK-2749 Bring monitored entity id and host into alert service on target detection.

### PR DESCRIPTION
**Pull request by Mohammad-Junid**
## Description
Just like node info , we need monitored entities info in alert service. Need to add monitored entities publisher in  service inventory, when monitored entity info is added, updated or deleted then need to reflect those changes  in the alert service. For implementation see NodeKafkaProducer in the inventory service. Create table with name of  simple_monitored_entity in the alert service, this table will have two columns monitored entity id and host.

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2749

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/AlertTestSteps.java [hunk 1]
```diff
@@ -528,4 +529,16 @@ public void prepareEventFromBelowThresholdDataWhenTheThresholdAlertStatusIs(
     public void sendEventThroughKafka() {
         kafkaTestHelper.sendToTopic(background.getEventTopic(), eventLog.toByteArray(), tenantSteps.getTenantId());
     }
+
+    @Then("verify that created alert is save with ManagedObjectType  as {string}")
```
Comment (aurang-zeb313) [inline alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/AlertTestSteps.java:533]: Replace save with saved

File: alert/src/test/resources/org/opennms/horizon/alertservice/node.feature [hunk 2]
```diff
@@ -90,3 +92,12 @@ Feature: Node operation feature
     Then Sent node for deletion using Kafka topic
     Then Verify node topic has 2 message for the tenant id "tenant-4"
     Then Verify node with id 155 is deleted
+
+
+    Scenario: Verify that monitord entity kafka topic recieve data
```
Comment (aurang-zeb313) [inline alert/src/test/resources/org/opennms/horizon/alertservice/node.feature:97]: Replace recieve with receive

File: alert/src/test/resources/org/opennms/horizon/alertservice/node.feature [hunk 3]
```diff
@@ -90,3 +92,12 @@ Feature: Node operation feature
...
+      Given Tenant id "tenant-5"
+      Given Monitored entity operation data
+        | id   | tenant_id | monitored_entity_id | host_name |
+        | 105  | tenant-5  | simple_entity_id  | www.google.com |
+      Given send monitored entity data to kafka topic
+      Then Verify monitored entity topic has 1 messages for the tenant with Operation type "UPDATE"
```
Comment (aurang-zeb313) [inline alert/src/test/resources/org/opennms/horizon/alertservice/node.feature:103]: replace messages with message

File: events/docker-it/src/test/resources/org/opennms/horizon/events/node.feature [hunk 4]
```diff
@@ -16,7 +16,7 @@ Feature: Node operation feature
       | 1  | tenant-1  | first  |
       | 2  | tenant-2  | second |
     And Sent node message to Kafka topic
-    Then Verify node topic has 2 message for node operation "UPDATE_NODE"
+    Then Verify node topic has 2 message for node operation "UPDATE"
```
Comment (aurang-zeb313) [inline events/docker-it/src/test/resources/org/opennms/horizon/events/node.feature:19]: replace message with messages

**Review by aurang-zeb313 — CHANGES_REQUESTED** [review]
Overall good work; there are just a few small changes to be made.

File: events/docker-it/src/test/resources/org/opennms/horizon/events/node.feature (hunk 4 above)
Comment (Mohammad-Junid) [reply events/docker-it/src/test/resources/org/opennms/horizon/events/node.feature]: Done

File: alert/src/test/resources/org/opennms/horizon/alertservice/node.feature (hunk 3 above)
Comment (Mohammad-Junid) [reply alert/src/test/resources/org/opennms/horizon/alertservice/node.feature]: Done

File: alert/src/test/resources/org/opennms/horizon/alertservice/node.feature (hunk 2 above)
Comment (Mohammad-Junid) [reply alert/src/test/resources/org/opennms/horizon/alertservice/node.feature:97]: Done

File: alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/AlertTestSteps.java (hunk 1 above)
Comment (Mohammad-Junid) [reply alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/AlertTestSteps.java]: Done

File: inventory/main/src/main/java/org/opennms/horizon/inventory/service/taskset/response/MonitorResponseService.java [hunk 5]
```diff
@@ -124,4 +134,11 @@ private void triggerEvent(
         var eventLog = EventLog.newBuilder().setTenantId(tenantId).addEvents(eventBuilder.build());
         eventProducer.sendEvent(eventLog.build());
     }
+
+    public void publishMonitoredEntity(MonitoredEntityStateDTO monitoredEntityStateDTO) {
```
Comment (cgorantla) [inline inventory/main/src/main/java/org/opennms/horizon/inventory/service/taskset/response/MonitorResponseService.java:138]: we shouldn't use response service for these operations.

 We should probably try to create monitored_entity_state with down status along with simple_monitored_entity

File: inventory/main/src/main/java/org/opennms/horizon/inventory/component/MonitorEntityKafkaProducer.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    @Value("${kafka.topics.monitor-entity:" + DEFAULT_TOPIC + "}")
+    private String monitoredEntityTopic;
+
+    private final KafkaTemplate<String, byte[]> kafkaTemplate;
+
+    public MonitorEntityKafkaProducer(KafkaTemplate<String, byte[]> kafkaTemplate) {
+        this.kafkaTemplate = kafkaTemplate;
+    }
+
+    public void publishTagUpdate(MonitoredEntityOperation monitoredEntityOperation) {
```
Comment (cgorantla) [inline inventory/main/src/main/java/org/opennms/horizon/inventory/component/MonitorEntityKafkaProducer.java:47]: these are not tags.
Also if we include NodeOperationProto and MonitoredEntityOperation in one common message, we can push these updates within same topic.  we don't need separate topic just for monitored entity ids.

File: alert/src/main/java/org/opennms/horizon/alertservice/grpc/AlertGrpcService.java [hunk 7]
```diff
@@ -162,49 +166,99 @@ public void listAlerts(ListAlertsRequest request, StreamObserver<ListAlertsRespo
...
                 }
-            } catch (NumberFormatException ex) {
-                // Just swallow
             }
+        } catch (NumberFormatException ex) {
+            // Just swallow
+        }
+    }
+
+    private void addMonitorEntityHostNameLabel(
```
Comment (cgorantla) [inline alert/src/main/java/org/opennms/horizon/alertservice/grpc/AlertGrpcService.java:208]: needs discussion on this. We may not need hostname from if we are carrying this in the event itself.

**Pull request closed without merging**